File: P13/src/mdp/Direktur.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package mdp;

import p13.Gaji;
import p13.Karyawan;

/**
 *
 * @author user
 */
public class Direktur extends Karyawan implements Gaji {
    private Double tunjangan;
    private Double gajipokok;

    public Double getTunjangan() {
        return tunjangan;
    }

    public void setTunjangan(Double tunjangan) {
        this.tunjangan = tunjangan;
    }
    
    public Direktur(String nama) {
        super.setNama(nama);
    }

    @Override
    public Double getTotal() {
        return this.getTunjangan() + this.getGajiPokok();
    }

    @Override
    public Double getGajiPokok() {
        return this.gajipokok;
    }

    @Override
    public void setGajiPokok(Double gajipokok) {
        this.gajipokok = gajipokok;
    }
    
}

File: p16/python/mdp/Direktur.py
from Gaji import Gaji
from Karyawan import Karyawan
from biodata import biodata

class Direktur(Karyawan, Gaji):
    #__private
    #_protected
    #public
    __tunjangan=0.0
    __gajipokok=0.0
    

    def __init__(self, tunjangan):
        self.__tunjangan = tunjangan

    def setTunjangan(self, tunjangan):
        self.__tunjangan =tunjangan
    
    def getTunjangan(self):
        return self.__tunjangan
    
    def setGajipokok(self, gajipokok):
        self.__gajipokok = gajipokok * 100
    
    def getGajiPokok(self):
        return self.__gajipokok

    def setNama(self, nama):
        self.nama=nama

    def getNama(self):
        return self.nama
    
